app/models/queue_item.rb

```diff
@@ -0,0 +1,151 @@
+class QueueItem < ActiveRecord::Base
+  belongs_to :user
+  belongs_to :video
+  before_validation :set_position, unless: :has_position?
+
+  delegate :category, to: :video
+
+  def rating
+    review.try(:rating)
+  end
+
+  def destroy_by_user(the_user)
+    if user == the_user
+      destroy
+      QueueItem.normalize(user)
+      true
+    else
+      false
+    end
+  end
+
+  def update_rating(rating)
+    if review
+      review.update_attributes(rating: rating)
+    else
+      video.reviews
+        .build(user_id: user.id, rating: rating)
+        .save(validate: false)
+    end
+  end
+
+  def review
+    video.reviews.find_by(user_id: user.id)
+  end
+
+  private
+
+  def set_position
+    self.position = QueueItem.where(user_id: self.user_id).count + 1
+  end
+
+  def has_position?
+    !!self.position
+  end
+end
+
+class << QueueItem
```

Comment:
Code from here till end of the file does not belong here. I am also not positive on what you want to achieve.